api/src/main/java/org/openmrs/module/auditlog/dao/impl/AuditLogDaoImpl.java

```diff
@@ -8,25 +8,27 @@
 import org.hibernate.sql.JoinType;
 import org.openmrs.Patient;
 import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
 import org.openmrs.module.auditlog.dao.AuditLogDao;
 import org.openmrs.module.auditlog.model.AuditLog;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
 
-@Component
 public class AuditLogDaoImpl implements AuditLogDao {
-    @Autowired
+
     private SessionFactory sessionFactory;
-    @Autowired
-    private PatientService patientService;
+
+    public void setSessionFactory(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
 
     protected static Integer LIMIT = 50;
 
+    PatientService patientService = Context.getPatientService();
```

Comment:
I am guessing by mistake? Dao should not have to use service

Reply:
true @angshu its wrongly used here. I just removed the Autowired annotation from the file and getting from Context. How shall we handle this ?

Reply:
I mean as part of this card or a separate card can be created to fix this ?

Reply:
ok. Sure. please create a separate card for this. patient ref should be passed from the service.